### PR TITLE
Add header and footer

### PR DESCRIPTION
**Pull request by EvonDong**

### CODE REVIEW

File: src/layouts/Footer.js [hunk 1]
```diff
@@ -1,14 +1,135 @@
 import React from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import Container from '@material-ui/core/Container';
+import Link from '@material-ui/core/Link';
+import { Paper, withStyles, Grid, TextField, Button, AppBar, Typography, Toolbar, CssBaseline} from '@material-ui/core';
```
Comment (C-likethis123) [inline src/layouts/Footer.js]: Possible to put all material-ui/core imports in the same line? Like this:

`import { Container, Link, Paper, withStyles, Grid, TextField, Button, AppBar, Typography, Toolbar, CssBaseline} from '@material-ui/core';`

File: src/layouts/Footer.js [hunk 2]
```diff
@@ -1,14 +1,135 @@
...
+    marginTop: theme.spacing(8),
+    marginBottom: theme.spacing(2),
+  },
+  footer: {
+    position: "fixed",
+    bottom: "0",
+    padding: theme.spacing(3, 2),
+    marginTop: 'auto',
+    backgroundColor:
+      theme.palette.type === 'light' ? theme.palette.grey[200] : theme.palette.grey[800],
```
Comment (C-likethis123) [inline src/layouts/Footer.js:37]: Do we have a light/dark mode? Think this is not the recommended way to implement light/dark mode as well, but I don't think we set up the light/dark mode theming yet.

File: src/layouts/Footer.js [hunk 3]
```diff
@@ -1,14 +1,135 @@
...
+                <Copyright />
+              </Container>
+            </footer>
+          </div>
+      </div>
+  )
+}
+}
+
+export default withStyles(useStyles)(Footer);
```
Comment (C-likethis123) [inline src/layouts/Footer.js:133]: Stick to either `makeStyles` or `withStyles` for theming.

File: src/layouts/Header.js [hunk 4]
```diff
@@ -1,14 +1,92 @@
...
+      <AppBar position="static">
+        <Toolbar>
+          <IconButton edge="start" className={classes.menuButton} color="inherit" aria-label="menu">
+            <MenuIcon />
+          </IconButton>
+          <Typography variant="h6" className={classes.title}>
+            Home | About
+          </Typography>
+          {auth && (
+            <div>
```
Comment (C-likethis123) [inline src/layouts/Header.js:51]: `Print` uses `Select` elements to create dropdown but this programmically opens/closes the dropdown? Think we should stick to one for consistency. I personally prefer `Select`, but it's open for discussion

File: src/layouts/Header.js [hunk 5]
```diff
@@ -1,14 +1,92 @@
...
+}));
+
 /**
  * For the header
  * @param {} props 
  */
-function Header(props) {
+
+export default function Header() {
+  const classes = useStyles();
```
Comment (C-likethis123) [inline src/layouts/Header.js:26]: Also this uses hooks while our project mainly uses class-based components? Should we switch to functional based components and hooks for consistency?

**Review by C-likethis123 — CHANGES_REQUESTED** [review]
Some issues which I think we should discuss about

File: src/layouts/Footer.js (hunk 1 above)
Comment (EvonDong) [reply src/layouts/Footer.js]: Fixed in the latest push

File: src/layouts/Footer.js (hunk 2 above)
Comment (EvonDong) [reply src/layouts/Footer.js:37]: Fixed in the latest push

File: src/layouts/Header.js (hunk 4 above)
Comment (EvonDong) [reply src/layouts/Header.js:51]: Fixed in latest push

**Pull request merged**